### PR TITLE
Release plugin

### PR DESCRIPTION
**Pull request by danyaljj**
(almost) the same as #372. 
Everything seems to be working now. The only missing piece of puzzle is activating `pushChanges`, which is commented out (didn't even try). Sth to try for future.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import _root_.sbtrelease.ReleasePlugin.autoImport.ReleaseStep
+import _root_.sbtrelease.ReleasePlugin.autoImport._
+import _root_.sbtrelease.ReleaseStateTransformations._
```
Comment (bhargav) [inline build.sbt]: Minor: You can replace all three imports with `import ReleaseStateTransformations._` ReleasePlugin and ReleaseStep are automatically imported by that.

File: build.sbt [hunk 2]
```diff
@@ -13,24 +16,48 @@ val headerMsg =  """/** This software is released under the University of Illino
                         |  */
                         |""".stripMargin
 
-lazy val root = (project in file(".")).
-  aggregate(saulCore, saulExamples)
-  .enablePlugins(AutomateHeaderPlugin)
+
+lazy val saulUser = System.getenv("SAUL_USER")
+lazy val user = if(saulUser == null) System.getProperty("user.name") else saulUser
```
Comment (bhargav) [inline build.sbt]: nit: single space after if 😋

**Comment by bhargav** [comment]
Changes look good. Minor comments. Please merge this change by selecting "Squash and Merge" from the Merge pull request submenu.

File: build.sbt (hunk 2 above)
Comment (danyaljj) [inline build.sbt]: hmm ... I think it already has single space?

Comment (bhargav) [inline build.sbt]: Actually I meant `if (saulUser` right after the keyword. Won't be that picky though.

Comment (danyaljj) [inline build.sbt]: I see I see ...

**Comment by danyaljj** [comment]
Applied the suggestions.

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
+
+// to automatically add header descriptions to files
 addSbtPlugin("de.heikoseeberger" % "sbt-header" % "1.6.0")
 
+// to bump the version numbers and automate the release process
+addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.3")
+
+// assembly plugin to create fat jars
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.1")
+
```
Comment (kordjamshidi) [inline project/plugins.sbt]: Do you think this is necessary?

Comment (danyaljj) [inline project/plugins.sbt]: The assembly plugin already existed (in fact you added: https://github.com/IllinoisCogComp/saul/blob/master/project/assembly.sbt#L2 ). I just moved it here next to other plugins. 

Currently we are not directly using it. However I don't think it's causing any computational overhead. So ok to keep or remove.

Comment (kordjamshidi) [inline project/plugins.sbt]: Oh, I see. Yes, I remembered I tried to use it for having stand alone executable on the servers from my home directory but building the fat jar was so slow and it was so huge that I found it useless after all.

Comment (danyaljj) [inline project/plugins.sbt]: I see ... I'll drop it then.

Comment (kordjamshidi) [inline project/plugins.sbt]: Ok, otherwise, it is ok for me to merge.

**Pull request merged**